### ISSUE TITLE
Configs to set `.vue` extension via `overrides`

### ISSUE THREAD

**Issue by privatenumber** [vuejs/eslint-plugin-vue#2036]
**The problem you want to solve.**
Currently, the docs recommend passing the `.vue` extension in via CLI: https://eslint.vuejs.org/user-guide/#usage

This is cumbersome to declare for each config because Vue SFCs are always in `.vue` files.

**Your take on the correct solution to problem.**
When an extension/pattern is set via `overrides`, the extension no longer needs to be manually passed into ESLint via the `--ext` flag.

I would like to recommend that configs use `overrides` so ESLint can automatically detect `.vue` files without the user needing to pass it in:

```js
{
	"overrides": [
		{
			"files": "*.vue",
			// ...
		}
	]
}
```

**Other info**
- Applying the plugin rules specifically to `.vue` extensions should make it easier to handle conflicts like this: https://github.com/vuejs/eslint-plugin-vue/blob/9673a61b08bdadf31b191e0dff30873678c7011b/docs/rules/script-indent.md?plain=1#L67

**Comment by ota-meshi** [comment]
Thank you for posting the issue,
I think maybe we can move only the parser config to `*.vue` since the rules may also check `*.js` files.

Since it's a breaking change, it should be done in a major version, but I don't know yet if it will be in the next major version.
Since the ESLint team is planning a new flat config, we may provide a completely different shareable config.
https://eslint.org/blog/2022/08/new-config-system-part-1/

**Comment by FloEdelmann** [comment]
Fixed by #2036 in v10.
